packages/ethereum/hardhat.config.ts

```diff
@@ -9,78 +9,75 @@ import 'hardhat-deploy'
-  ENVIRONMENT_ID = 'default'
-} = process.env
-const GAS_MULTIPLIER = 1.1
+const { DEPLOYER_WALLET_PRIVATE_KEY, ETHERSCAN_KEY, HOPR_ENVIRONMENT_ID } = process.env
+import { expandVars } from '@hoprnet/hopr-utils'
+
+const PROTOCOL_CONFIG = require('../core/protocol-config.json')
 
 extendConfig((config: HardhatConfig) => {
   config.etherscan.apiKey = ETHERSCAN_KEY
 })
 
 extendEnvironment((hre: HardhatRuntimeEnvironment) => {
-  hre.environment = ENVIRONMENT_ID
+  hre.environment = HOPR_ENVIRONMENT_ID
 })
 
-const hardhatConfig: HardhatUserConfig = {
-  defaultNetwork: 'hardhat',
-  networks: {
-    // hardhat-deploy cannot run deployments if the network is not hardhat
-    // we use an ENV variable (which is specified in our NPM script)
-    // to let hardhat know we want to run hardhat in 'development' mode
-    // this essentially enables mining, see below
-    hardhat: {
-      live: false,
-      tags: [DEVELOPMENT ? 'development' : 'testing'] as NetworkTag[],
-      saveDeployments: true,
-      mining: DEVELOPMENT
-        ? {
-            auto: true, // every transaction will trigger a new block (without this deployments fail)
-            interval: [1000, 3000] // mine new block every 1 - 3s
-          }
-        : undefined
-    },
-    goerli: {
-      ...networks.goerli,
-      live: true,
-      tags: ['staging'] as NetworkTag[],
-      gasMultiplier: GAS_MULTIPLIER + 0.3, // Görli has been failing lately with underpriced txs
-      url: `https://goerli.infura.io/v3/${INFURA_KEY}`,
-      accounts: DEPLOYER_WALLET_PRIVATE_KEY ? [DEPLOYER_WALLET_PRIVATE_KEY] : []
-    },
-    xdai: {
-      ...networks.xdai,
-      live: true,
-      tags: ['development'] as NetworkTag[],
-      gasMultiplier: GAS_MULTIPLIER,
-      url: `https://provider-proxy.hoprnet.workers.dev/xdai_mainnet`,
-      accounts: DEPLOYER_WALLET_PRIVATE_KEY ? [DEPLOYER_WALLET_PRIVATE_KEY] : []
-    },
-    mumbai: {
-      ...networks.mumbai,
-      live: true,
-      tags: ['development'] as NetworkTag[],
-      gasMultiplier: GAS_MULTIPLIER,
-      url: `https://polygon-mumbai.infura.io/v3/${INFURA_KEY}`,
-      accounts: DEPLOYER_WALLET_PRIVATE_KEY ? [DEPLOYER_WALLET_PRIVATE_KEY] : []
-    },
-    polygon: {
-      ...networks.polygon,
-      live: true,
-      tags: ['development'] as NetworkTag[],
-      url: `https://polygon-mainnet.infura.io/v3/${INFURA_KEY}`,
-      accounts: DEPLOYER_WALLET_PRIVATE_KEY ? [DEPLOYER_WALLET_PRIVATE_KEY] : []
+// For reference on how the configuration is structured refer to:
+//
+// https://hardhat.org/hardhat-network/reference/#config
+// https://github.com/wighawag/hardhat-deploy/blob/master/README.md
+function networkToHardhatNetwork(name: String, input: any): any {
+  let cfg: any = {
+    chainId: input.chain_id,
+    gasMultiplier: input.gas_multiplier,
+    live: input.live,
+    tags: input.tags,
+    // used by hardhat-deploy
+    saveDeployments: true,
+    mining: undefined
+  }
+
+  if (input.gas) {
```

Comment:
So here `input.gas` defines "gasPrice". Is `gas` (which defines "gas limit") handled somewhere? Or we don't care about it and just use the default "block gas limit"?

Reply:
good point, let's address that after merging the PR

Reply:
Tracked: https://github.com/hoprnet/hoprnet/issues/2859